src/Sylius/Bundle/UserBundle/Doctrine/ORM/UserRepository.php

```diff
@@ -41,10 +39,11 @@ public function createFilterPaginator($criteria = array(), $sorting = array(), $
 
         if (isset($criteria['query'])) {
             $queryBuilder
-                ->where('o.username LIKE :query')
-                ->orWhere('o.email LIKE :query')
-                ->orWhere('o.firstName LIKE :query')
-                ->orWhere('o.lastName LIKE :query')
+                ->leftJoin($this->getAlias().'.customer', 'customer')
+                ->where('customer.emailCanonical LIKE :query')
+                ->orWhere('customer.firstName LIKE :query')
+                ->orWhere('customer.lastName LIKE :query')
+                ->orWhere($this->getAlias().'.username LIKE :query')
                 ->setParameter('query', '%'.$criteria['query'].'%')
```

Comment:
Just a point of note, these queries can be painfully slow with large customer sets (with mysql anyway) due to the use of full wildcards at both ends %X%, if you use just a training wildcard match then mysql can use indexes (if any are defined).

Probably not worth changing now but we've got over a million customers so I'll let you know how it performs ;)